jax/_src/numpy/reductions.py

```diff
@@ -72,9 +72,13 @@ def _promote_integer_dtype(dtype: DTypeLike) -> DTypeLike:
   if dtypes.issubdtype(dtype, np.bool_):
     return dtypes.int_
   elif dtypes.issubdtype(dtype, np.unsignedinteger):
+    if dtype in [np.dtype('uint4'), np.dtype('uint2')]:
```

Comment:
I think there's a better fix. Try removing this change and instead changing the code below to use `ml_dtypes.iinfo`.


Add a test?

Reply:
Agreed – `jnp.iinfo` is an alias that will prevent you having to explicitly import `ml_dtypes`.

When this code was written, there were no custom integer types.